SYCL/ESIMD/api/functional/ctors/operator_assignment_copy_core.cpp

```diff
@@ -0,0 +1,118 @@
+//==------- operator_assignment_copy_core.cpp  - DPC++ ESIMD on-device test
+//          ----------------------------------------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu, level_zero
+// XREQUIRES: gpu
+// TODO gpu and level_zero in REQUIRES due to only this platforms supported yet.
+// The current "REQUIRES" should be replaced with "gpu" only as mentioned in
+// "XREQUIRES".
+// UNSUPPORTED: cuda, hip
+// RUN: %clangxx -fsycl %s -fsycl-device-code-split=per_kernel -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+//
+// Test for simd copy assignment operator.
+// The test creates source simd instance with reference data and invokes copy
+// assignment operator. It is expected that simd object that called copy
+// assignment operator will stor a bitwise same data as in the source simd
+// object.
+
+#include "common.hpp"
+
+using namespace sycl::ext::intel::experimental::esimd;
+using namespace esimd_test::api::functional;
+
+// Descriptor class for the case of calling constructor in initializer context.
```

Comment:
There is no initializer context below. Anyway, looks like we don't have a `context` for assignment operators.
As a suggestion, what if we have:

- cpp file with `copy_assignment`
- cpp file with `move_assignment`
- header file with generic `run_test` for move and copy assignment

Would it be reasonable? Given that:

- it would be great to check that `was_move_destination() == false` for copy assignment
- we might want someday to check copy assignment doesn't invalidate source simd, but we cannot allow such check for move assignment

Reply:
I've done it in this commit: c08d8bcd8cfb2adfcd4af2321500e8672a2b471c